File: MyPython/Binomial_Recursive.py
# AmirAnsarpour
# 40022830117248
# https://github.com/AmirAnsarpour

import time

def binomial_recursive(n, k):
    if k == 0 or k == n:
        return 1
    else:
        return binomial_recursive(n-1, k-1) + binomial_recursive(n-1, k)

n = 30
k = 20
start_time = time.time()
result = binomial_recursive(n, k)
end_time = time.time()

execution_time = "{:.10f}".format(end_time - start_time)


print(f"C({n}, {k}) = {result}")
print(f"Execution time: {execution_time} Secend.")

File: C++/Binomial_Recursive.cpp
// AmirAnsarpour
// 40022830117248
// https://github.com/AmirAnsarpour
// C++

#include <iostream>
#include <ctime>

int binomial_recursive(int n, int k) {
    if (k == 0 || k == n) {
        return 1;
    } else {
        return binomial_recursive(n - 1, k - 1) + binomial_recursive(n - 1, k);
    }
}

int main() {
    int n = 30;
    int k = 20;

    clock_t start_time = clock();
    int result = binomial_recursive(n, k);
    clock_t end_time = clock();

    double execution_time = static_cast<double>(end_time - start_time) / CLOCKS_PER_SEC;

    std::cout << "C(" << n << ", " << k << ") = " << result << std::endl;
    std::cout << "Execution time: " << execution_time << " seconds." << std::endl;

    return 0;
}
